api/src/plugins/plugins.service.ts

```diff
@@ -35,7 +44,37 @@ export class PluginService<T extends BasePlugin = BasePlugin> {
     Object.keys(PluginType).map((t) => [t as PluginType, new Map()]),
   );
 
-  constructor() {}
+  constructor(
+    private readonly settingService: SettingService,
+    private readonly logger: LoggerService,
+  ) {}
+
+  async onApplicationBootstrap(): Promise<void> {
+    await this.cleanup();
+  }
+
+  /**
+   * Cleanups the unregisterd plugins from settings.
+   *
+   */
+  async cleanup(): Promise<void> {
+    const activePlugins = this.getAll().map((handler) =>
+      handler.getNamespace(),
+    );
+
+    const pluginSettings = (await this.settingService.getAllGroups()).filter(
+      (group) => group.split('_').pop() === 'plugin',
+    ) as HyphenToUnderscore<PluginName>[];
+
+    const orphanSettings = pluginSettings.filter(
+      (group) => !activePlugins.includes(group),
+    );
+
+    for (const group of orphanSettings) {
+      this.logger.log(`Deleting orphaned settings for ${group}...`);
+      await this.settingService.deleteGroup(group);
+    }
+  }
```

Comment:
```suggestion
  async cleanup(): Promise<void> {
  const activePlugins =this.getAll().map((handler) => handler.getNamespace());

  const pluginSettings = await this.settingService.getExtensionSettings('plugin');
  const orphanSettings = pluginSettings.filter(
    (namespace) => !activePlugins.includes(namespace)
  );

  await Promise.all(
    orphanSettings.map(async (group) => {
      this.logger.log(`Deleting orphaned settings for ${group}...`);
      return this.settingService.deleteGroup(group);
    })
  );
}
```

getExtensionSettings('plugin') would retrieve using a mongo query that searches for groups that end with "_plugin".